benchmarks/synchrobench/src/main/java/com/yahoo/oak/synchrobench/MyBuffer.java

```diff
@@ -37,70 +36,46 @@ public int compareTo(MyBuffer o) {
         return compareBuffers(this, o);
     }
 
-    public static void serialize(MyBuffer inputBuffer, OakUnsafeDirectBuffer targetBuffer) {
-        serialize(inputBuffer, targetBuffer.getByteBuffer(), targetBuffer.getOffset());
-    }
-
-    public static void serialize(MyBuffer inputBuffer, ByteBuffer targetBuffer, int targetPos) {
+    public static void serialize(MyBuffer inputBuffer, OakScopedWriteBuffer targetBuffer) {
         // In the serialized buffer, the first integer signifies the size.
-        targetBuffer.putInt(targetPos, inputBuffer.capacity);
-        // Thus, the data position starts after the first integer.
-        targetPos += Integer.BYTES;
-
+    	targetBuffer.putInt(0,  inputBuffer.capacity);
         OakIntBufferSerializer.copyBuffer(inputBuffer.buffer, DATA_POS, inputBuffer.capacity / Integer.BYTES,
-                targetBuffer, targetPos);
-    }
-
-    public static MyBuffer deserialize(OakUnsafeDirectBuffer inputBuffer) {
-        return deserialize(inputBuffer.getByteBuffer(), inputBuffer.getOffset());
+        		targetBuffer, Integer.BYTES);
     }
 
-    public static MyBuffer deserialize(ByteBuffer inputBuffer, int inputPos) {
-        // In the serialized buffer, the first integer signifies the size.
-        int capacity = inputBuffer.getInt(inputPos);
-        // Thus, the data position starts after the first integer.
-        inputPos += Integer.BYTES;
-
+    public static MyBuffer deserialize(OakScopedReadBuffer inputBuffer) {
+        int capacity = inputBuffer.getInt(0);
         MyBuffer ret = new MyBuffer(capacity);
-        OakIntBufferSerializer.copyBuffer(inputBuffer, inputPos, capacity / Integer.BYTES, ret.buffer, DATA_POS);
+        OakIntBufferSerializer.copyBuffer(inputBuffer, Integer.BYTES, capacity / Integer.BYTES, ret.buffer, DATA_POS);
         return ret;
     }
 
+
     public static int compareBuffers(ByteBuffer buff1, int pos1, int cap1, ByteBuffer buff2, int pos2, int cap2) {
         return OakIntBufferComparator.compare(buff1, pos1, cap1 / Integer.BYTES,
                 buff2, pos2, cap2 / Integer.BYTES);
     }
 
-    public static int compareBuffers(OakUnsafeDirectBuffer buffer1, OakUnsafeDirectBuffer buffer2) {
-        ByteBuffer buf1 = buffer1.getByteBuffer();
-        ByteBuffer buf2 = buffer2.getByteBuffer();
-
-        int pos1 = buffer1.getOffset();
-        int pos2 = buffer2.getOffset();
+    public static int compareBuffers(ByteBuffer buff1, int pos1, int cap1, OakScopedReadBuffer buff2, int pos2, int cap2) {
```

Comment:
I am a bit lost here... Can you please add some comments? Why so many `compareBuffers`? Why are they all public?

Reply:
what we had initally is just comparing bytebuffers, which is still exposed 
I added the option to compare ScopedReadBuffer